__init__.py

```diff
@@ -360,18 +360,40 @@ def selectLanguage( self, path ):
 		elif conf["viur.languageMethod"] == "url":
 			tmppath = urlparse.urlparse( path ).path
 			tmppath = [ urlparse.unquote( x ) for x in tmppath.lower().strip("/").split("/") ]
+			langSet = False
 			if len( tmppath )>0 and tmppath[0] in conf["viur.availableLanguages"]+list( conf["viur.languageAliasMap"].keys() ):
 				self.language = tmppath[0]
-				return( path[ len( tmppath[0])+1: ] ) #Return the path stripped by its language segment
+				langSet = True
+				return path[ len( tmppath[0])+1: ]  # Return the path stripped by its language segment
 			else: # This URL doesnt contain an language prefix, try to read it from session
 				if session.current.getLanguage():
 					self.language = session.current.getLanguage()
-				elif "X-Appengine-Country" in self.request.headers.keys():
+					langSet = True
+				if not langSet and "Accept-Language" in self.request.headers:
```

Comment:
The control-flow is botched. Instead of carrying the marker langSet, just return path in L371. Same thing at the end of this if block - if found return path.